Code/Framework/AzToolsFramework/AzToolsFramework/ComponentMode/ComponentModeCollection.h

```diff
@@ -107,6 +108,7 @@ namespace AzToolsFramework
             bool ActiveComponentModeChanged(const AZ::Uuid& previousComponentType);
 
             AZStd::vector<AZ::Uuid> m_activeComponentTypes; ///< What types of ComponentMode are currently active.
+            AZStd::vector<AZStd::pair<AZ::EntityComponentIdPair, AZ::Uuid>> m_activeComponentModes; ///< What Entity Components have a ComponentMode currently active.
```

Comment:
Yes and no. `m_activeComponentTypes` will only store the types of the component modes, `m_activeComponentModes` keeps track of all components with active component modes. In the case of multi-entity editing, the latter will have more entries since more component modes are created for a single type.
I originally was going for just expanding `m_activeComponentTypes` but it was bloating up the scope significantly, so yes we could refactori this setup but I would probably want a better view of the whole architecture before touching it :)

Reply:
Okay understood thanks! Follow-up makes sense! 😅